needle.pm

```diff
@@ -195,7 +202,9 @@ sub wanted_ {
 }
 
 sub init {
-    $needledir = shift if @_;
+    $needledir    = shift if @_;
+    $shared_cache = shift if @_;
```

Comment:
i changed it to be  `($needledir, $shared_cache) = @_; ` $shared_cache will be undefined anyways, and no harm will come if the cache is not enabled.